Add filter_path parameter and restructure article

smsalamon opened the pull request:
Signed-off-by: smsalamon <stacey.salamon@gmail.com>

### Description
Add filter_path to list of common REST parameters and restructure page as suggested in the Issue to make the descriptions easier to read.

### Issues Resolved
Resolves https://github.com/opensearch-project/documentation-website/issues/519.

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Naarcha-AWS commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,26 @@ nav_order: 93
-Content type | To specify the type of content in the request body, use the `Content-Type` key name in the request header. Most operations support JSON, YAML, and CBOR formats. | `POST _scripts/<template_name> -H 'Content-Type: application/json`
-Request body in query string | If the client library does not accept a request body for non-POST requests, use the `source` query string parameter to pass the request body. Also, specify the `source_content_type` parameter with a supported media type such as `application/json`. | `GET _search?source_content_type=application/json&source={"query":{"match_all":{}}}`
-Stack traces | To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL. | `GET <index_name>/_search?error_trace=true`
+## Human-readable output
```
Can we add a line-break between each header and example?

smsalamon replied:
Done. 👍🏻

Naarcha-AWS reviewed (commented):
Looks good, just one small comment.

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,32 @@ nav_order: 93
+## Human-readable output
+
+To convert output units to human-readable values (for example, `1h` for 1 hour and `1kb` for 1,024 bytes), add `?human=true` to the request URL.  
+Example: `GET <index_name>/_search?human=true`
```
```suggestion
#### Sample request

The following request requires that responses values should be in human-readable format:

GET <index_name>/_search?human=true
```

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,32 @@ nav_order: 93
+## Pretty result
+
+To get back JSON responses in a readable format, add `?pretty=true` to the request URL.  
+Example: `GET <index_name>/_search?pretty=true`
```
```suggestion

 ####  Sample request
 
 The following request requires the response to display in pretty JSON format:
 
 ```json
  GET <index_name>/_search?pretty=true
```

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,32 @@ nav_order: 93
+## Content type
+
+To specify the type of content in the request body, use the `Content-Type` key name in the request header. Most operations support JSON, YAML, and CBOR formats.  
+Example: `POST _scripts/<template_name> -H 'Content-Type: application/json`
```
```suggestion

#### Sample request

The following request specifies JSON  format for the request body:

```json
POST _scripts/<template_name> -H 'Content-Type: application/json
```

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,32 @@ nav_order: 93
+## Request body in query string
+
+If the client library does not accept a request body for non-POST requests, use the `source` query string parameter to pass the request body. Also, specify the `source_content_type` parameter with a supported media type such as `application/json`.  
+Example: `GET _search?source_content_type=application/json&source={"query":{"match_all":{}}}`
```
```suggestion

#### Sample request

The following request searches documents in the `shakespeare` index for a specific field and value:

```json
GET shakespeare/_search?source={"query":{"exists":{"field":"speaker"}}}&source_content_type=application/json
```

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,32 @@ nav_order: 93
+## Stack traces
+
+To include the error stack trace in the response when an exception is raised, add `error_trace=true` to the request URL.  
+Example: `GET <index_name>/_search?error_trace=true` 
```
```suggestion

#### Sample request

The following request sets `error_trace` to `true` so that the response will return exception-triggered errors:

```json
GET <index_name>/_search?error_trace=true
```

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,32 @@ nav_order: 93
+## Filtered responses
+
+To reduce the response size use the `filter_path` parameter to filter the fields that are returned. This parameter takes a comma-separated list of filters. It supports using wildcards to match any field or part of a field's name. You can also exclude fields with `-`.  
+Example: `GET _search?filter_path=<field_name>.*,-<field_name>`
```
```suggestion

#### Sample request

The following request specifies filters to limit the fields returned in the response:

```json
GET _search?filter_path=<field_name>.*,-<field_name>
```

alicejw-aws reviewed (approved):
This is a good improvement for readability! 

I made a couple of suggestions to modify the examples to match our style formats outlined in the [API reference page template]9https://github.com/opensearch-project/documentation-website/blob/main/STYLE_API_TEMPLATE.md#sample-request).

I used the in-line comment tool to be precise, but you can make updates directly from your branch instead of selecting them with "commit suggestion."

**Note:** One of the query examples did not test properly, so I modified it with one that returned "200-OK" in the Console.

Thanks and keep up the good work! We appreciate your contributions.

smsalamon commented:
> This is a good improvement for readability!
> 
> I made a couple of suggestions to modify the examples to match our style formats outlined in the [API reference page template](https://github.com/opensearch-project/documentation-website/blob/main/STYLE_API_TEMPLATE.md#sample-request).
> 
> I used the in-line comment tool to be precise, but you can make updates directly from your branch instead of selecting them with "commit suggestion."
> 
> **Note:** One of the query examples did not test properly, so I modified it with one that returned "200-OK" in the Console.
> 
> Thanks and keep up the good work! We appreciate your contributions.

Thanks for reviewing @alicejw-aws and for sharing the API template! I've made the requested changes.

Also, I only just noticed that line 47 appears to be missing a closing single quote for the content_type parameter. I've added it now, but can you confirm that is the correct syntax?

alicejw-aws commented on _opensearch/common-parameters.md:
````diff
@@ -8,10 +8,81 @@ nav_order: 93
+
+```json
+
+POST _scripts/<template_name> -H 'Content-Type: application/json'
````
Hi @smsalamon , yes! good catch! This looks like an HTTP POST request, but then they've added 
one of the cURL arguments to the end of the request. 

curl -XPUT -k -H 'Content-Type: application/json'
I need to do a bit of testing and will suggest either we replace it with a proper cURL command, or remove the incorrect request. thanks

alicejw-aws replied:
```suggestion
POST _scripts/<template_name>?content-type= application/json
```

alicejw-aws replied:
```suggestion
curl -H "Content-type: application/json" -XGET localhost:9200/_scripts/<template_name>
```

alicejw-aws commented on _opensearch/common-parameters.md:
```diff
@@ -8,10 +8,81 @@ nav_order: 93
+
+### Sample request
+
+The following request specifies JSON format for the request body:
```
```suggestion
The following cURL request specifies JSON format for the request body:
```

alicejw-aws reviewed (commented):
Hi @smsalamon, we got an example cURL request from @kolchfa-aws that we could replace here. otherwise, looks like it's good to go.
Thanks

smsalamon commented:
> Hi @smsalamon, we got an example cURL request from @kolchfa-aws that we could replace here. otherwise, looks like it's good to go. Thanks

Thanks so much @alicejw-aws and @kolchfa-aws!
